File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/GetWorkAreaUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.result.WorkAreaResult;
import reactor.core.publisher.Mono;

public interface GetWorkAreaUseCase {

    Mono<WorkAreaResult> getById(Long workAreaId);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/GetWorkAreaUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.result.WorkAreaResult
import reactor.core.publisher.Mono

interface GetWorkAreaUseCase {
    fun getById(workAreaId: Long): Mono<WorkAreaResult>
}
